### ISSUE TITLE
Unable to connect from Velocity

### ISSUE THREAD

**Issue by DBTDerpbox** [Declipsonator/Chat-Control#3]
I currently have a Fabric server using FabricProxy-Lite to connect to a [Velocity Proxy](https://velocitypowered.com/).

With FabricProxy-Lite disabled and connecting directly to the fabric server, i am able to connect and play just fine, but when trying to connect from Velocity with FabricProxy-Lite enabled, I fail to connect. This issue is not present with Chat Control not on the fabric server.

Here is the error Velocity prints out upon trying to connect:

```
[04:46:39 INFO]: [server connection] DBTDerpbox -> battle has connected
[04:46:40 ERROR]: [server connection] DBTDerpbox -> battle: exception encountered in com.velocitypowered.proxy.connection.backend.BackendPlaySessionHandler@1609f364
io.netty.handler.codec.CorruptedFrameException: Error decoding class com.velocitypowered.proxy.protocol.packet.AvailableCommands Direction CLIENTBOUND Protocol 1.18 State PLAY ID 12
        at com.velocitypowered.proxy.protocol.netty.MinecraftDecoder.handleDecodeFailure(MinecraftDecoder.java:128) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at com.velocitypowered.proxy.protocol.netty.MinecraftDecoder.tryDecode(MinecraftDecoder.java:83) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at com.velocitypowered.proxy.protocol.netty.MinecraftDecoder.channelRead(MinecraftDecoder.java:58) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:327) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:314) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:435) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:279) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:487) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:385) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:995) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.lang.IllegalArgumentException: Argument type identifier minecraft:char unknown.
        at com.velocitypowered.proxy.protocol.packet.brigadier.ArgumentPropertyRegistry.deserialize(ArgumentPropertyRegistry.java:75) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at com.velocitypowered.proxy.protocol.packet.AvailableCommands.deserializeNode(AvailableCommands.java:208) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at com.velocitypowered.proxy.protocol.packet.AvailableCommands.decode(AvailableCommands.java:86) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        at com.velocitypowered.proxy.protocol.netty.MinecraftDecoder.tryDecode(MinecraftDecoder.java:81) ~[velocity.jar:3.1.2-SNAPSHOT (git-377e6b66-b140)]
        ... 26 more
[04:46:40 INFO]: [server connection] DBTDerpbox -> battle has disconnected
```

My Fabric server is on 1.18.2, with Fabric Loader 0.14.6

**Comment by Declipsonator** [comment]
Pretty sure I know whats happening here but I don't really have the time to fix it until friday.

**Comment by DBTDerpbox** [comment]
i'm sorry to bother with this, but is there any update on this?

**Comment by Declipsonator** [comment]
Oh I just had a bunch of exams and forgot about it, I'll make sure to fix it this weekend.

**Comment by Declipsonator** [comment]
Fixed with [c7ed088](https://github.com/Declipsonator/Chat-Control/commit/c7ed08879ead0f491cf614bdb892c7f253d1373e)

**Comment by Declipsonator** [comment]
https://modrinth.com/mod/chat/version/1.2
